kvbc/include/blockchain_type.hpp

```diff
@@ -0,0 +1,26 @@
+// Concord
+//
+// Copyright (c) 2022-2023 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <variant>
+#include <optional>
+#include <memory>
+#include "categorization/kv_blockchain.h"
+
+namespace concord::kvbc {
+enum BLOCKCHAIN_VERSION { CATEGORIZED_BLOCKCHAIN = 1, NATURAL_BLOCKCHAIN = 4 };
```

Comment:
Version or type?

Reply:
I think this is the version. For the same type of blockchain we will have different versions and they will coexist. 
A future use case can be said to implement categorized blockchain on top of Redis or aerospike as they may provide some feature that is better than rocksdb. Or implement a new type of blockchain on cockroach DB and mongo DB. One is a new-SQL DB and another is a no-SQL DB. In such scenarios for better performance, we might create a different KV Blockchain version for the same type and do A/B testing to finalize on default. 
There will be use case where these different versions co-exist, say vendor 1 likes to have blockchain in MongoDB and vendor 2 want it to be categorized as they definitely need Merkel tree. In such cases, we might provide the same type of KV blockchain with a new version.